app/controllers/stages_controller.rb

```diff
@@ -133,9 +129,8 @@ def stage_params
       :update_github_pull_requests,
       :comment_on_zendesk_tickets,
       command_ids: [],
-      flowdock_flows_attributes: [:id, :name, :token, :_destroy],
-      new_relic_applications_attributes: [:id, :name, :_destroy]
-    )
+      new_relic_applications_attributes: [:id, :name, :_destroy],
```

Comment:
I don't think iframes will work since we want to submit a form with
additional fields and not have multiple indepdent forms + multiple
independent controllers.

Users have to trust the plugins code they install, so they might as well
trust their migrations. The changes it will do are visible in development
since it modified the schema.rb, so there should not be a "oops my column
is gone" moment.

A separate plugin language/framework also means more work for us to write
it + more work to use it (I assume most plugins will be written by us) +
reinventing rails things / forcing people to learn new tools for
questionable gain down the line.

It might be simplest to initially go with the engine solution and then see
if we can refactor the extracted plugins to a nicer framework where it
makes sense/we see a common pattern. This way we will see what things are
needed/reusable before building a framework and always have the escape
hatch of just using rails.

On Wed, Mar 4, 2015 at 9:50 AM, Michael Grosser <mgrosser@zendesk.com>
wrote:

> meh ... ok :)
>
> On Wed, Mar 4, 2015 at 6:34 AM, Shane Hender <notifications@github.com>
> wrote:
>
>> In app/controllers/stages_controller.rb
>> <https://github.com/zendesk/samson/pull/259#discussion_r25778104>:
>>
>> > @@ -133,9 +129,8 @@ def stage_params
>> >        :update_github_pull_requests,
>> >        :comment_on_zendesk_tickets,
>> >        command_ids: [],
>> > -      flowdock_flows_attributes: [:id, :name, :token, :_destroy],
>> > -      new_relic_applications_attributes: [:id, :name, :_destroy]
>> > -    )
>> > +      new_relic_applications_attributes: [:id, :name, :_destroy],
>>
>> I saw your justification for this on your earlier PR, but it does break
>> with the ruby style guide:
>> https://github.com/bbatsov/ruby-style-guide#no-trailing-params-comma
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/zendesk/samson/pull/259/files#r25778104>.
>>
>
>